doc/user/client.md

```diff
@@ -275,9 +275,9 @@ func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, e
 	...
 
 	ctx := context.TODO()
-	dep.Spec.Selector.MatchLabels["is_running"] = "true"
 	// A merge patch will preserve other fields modified at runtime.
-	patch := client.MergeFrom(dep)
+	patch := client.MergeFrom(dep.DeepCopy())
```

Comment:
Why it is required? What issue do you face when you did not make a copy of the object?

Reply:
When following the current documentation, there was no change that took place. After using a copy instead, as seen in some of the client unit tests, everything worked as expected.

The working example was taken from the client tests:
```
By("patching the status of Deployment")
dep.SetGroupVersionKind(depGvk)
depPatch := client.MergeFrom(dep.DeepCopy())
dep.Status.Replicas = 1
err = cl.Status().Patch(context.TODO(), dep, depPatch)
Expect(err).NotTo(HaveOccurred())
```
Looking it over again, the actual culprit may have just been the incorrect ordering of the code in the documentation (the `client.MergeFrom` calls are happening after the object has already been changed)

Reply:
I think that the problem here is the order as well. 
Could you please confirm in your test and then update this PR with the correction?

Reply:
I tried with only the ordering fix, and the patch didn't apply any changes. Switched back to `DeepCopy`, and it worked again.

I should have also included a link to the upstream tests that I got the working example from: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client_test.go#L704

Reply:
Really tks for the collab and check. 👍   I believe that is great to add the controller-runtime links and references as always as possible. However, in this case, I am not so sure since it goes to the tests of the project. 

So, let's move forward and if you find after any reference to the docs which could, for example, be used to replace this example would be terrific.